### PR TITLE
Added documentation for Cloudflare DNS

### PR DESCRIPTION
**Pull request by SayakMukhopadhyay**
**What this PR does / why we need it**:
Adds documentation on how to create API keys for cloudflare DNS and its use in secrets.
**Special notes for your reviewer**:
None
**Release note**:

```improvement operator

```

### CODE REVIEW

File: examples/20-secret-cloudflare-credentials.yaml [hunk 1]
```diff
@@ -8,4 +8,5 @@ data:
   # replace '...' with the API Key generated. The token should have permission of Zone:Read and DNS:Edit for all zones.
   # You can optionally exclude certain zones
   # see https://support.cloudflare.com/hc/en-us/articles/200167836-Managing-API-Tokens-and-Keys
+  # For details see https://github.com/SayakMukhopadhyay/external-dns-management/blob/master/doc/cloudflare/README.md#using-the-api-key
```
Comment (MartinWeindel) [inline examples/20-secret-cloudflare-credentials.yaml:11]: Please adjust the URLs `https://github.com/SayakMukhopadhyay/external-dns-management/...`

File: examples/30-provider-cloudflare.yaml [hunk 2]
```diff
@@ -1,3 +1,4 @@
+# For details see https://github.com/SayakMukhopadhyay/external-dns-management/blob/master/doc/cloudflare/README.md
```
Comment (MartinWeindel) [inline examples/30-provider-cloudflare.yaml:1]: Same here, adjust the URL

File: examples/20-secret-cloudflare-credentials.yaml (hunk 1 above)
Comment (SayakMukhopadhyay) [reply examples/20-secret-cloudflare-credentials.yaml]: Fixed the issue.

File: examples/30-provider-cloudflare.yaml (hunk 2 above)
Comment (SayakMukhopadhyay) [reply examples/30-provider-cloudflare.yaml]: Fixed this too!

**Comment by SayakMukhopadhyay** [comment]
I have also added some fixes regarding some confusion over the use of `key` and `token`.

File: doc/cloudflare/README.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+To generate the token make sure the token has permission of Zone:Read and DNS:Edit for 
+all zones. Optionally you can exclude certain zones.
+
+**Note: You need to `Include` `All zones` in the `Zone Resources` section. Setting 
+`Specific zone` doesn't work. But you can still add one or more `Exclude`s.**
+
+![API token creation](api-token-creation.png)
+
+Generate the token and keep this key safe as it won't be shown again.
+
```
Comment (vasu1124) [inline doc/cloudflare/README.md:22]: Then base64 convert the token, e.g.
```
$ echo -n '1234567890123456789' | base64
MTIzNDU2Nzg5MDEyMzQ1Njc4OQ==
```

File: doc/cloudflare/README.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+## Using the API Token
+
```
Comment (vasu1124) [inline doc/cloudflare/README.md:24]: Create a secret with the base64 encoded token:

File: doc/cloudflare/README.md [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: cloudflare-credentials
+  namespace: default
+type: Opaque
+data:
+  CLOUDFLARE_API_TOKEN: 1234567890123456789
```
Comment (vasu1124) [inline doc/cloudflare/README.md:33]: CLOUDFLARE_API_TOKEN: MTIzNDU2Nzg5MDEyMzQ1Njc4OQ==

File: doc/cloudflare/README.md [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+```
```
Comment (vasu1124) [inline doc/cloudflare/README.md:34]: **Note: If you are delegating the DNS Management to Gardener, ensure that you annotate the above secret in your project namespace with `dns.gardener.cloud/class: garden`. **

Comment (SayakMukhopadhyay) [reply doc/cloudflare/README.md:34]: Is this true only for this provider or for all providers?

File: doc/cloudflare/README.md (hunk 3 above)
Comment (SayakMukhopadhyay) [reply doc/cloudflare/README.md:22]: Added the same

File: doc/cloudflare/README.md (hunk 4 above)
Comment (SayakMukhopadhyay) [reply doc/cloudflare/README.md:32]: Added explanation on how to create the secret

File: doc/cloudflare/README.md (hunk 5 above)
Comment (SayakMukhopadhyay) [reply doc/cloudflare/README.md]: Fixed the value with the base64 one.

File: doc/cloudflare/README.md (hunk 6 above)
Comment (MartinWeindel) [reply doc/cloudflare/README.md:45]: that's true for all providers

Comment (SayakMukhopadhyay) [reply doc/cloudflare/README.md:45]: Then I think this note should be placed in the main README.

**Comment by MartinWeindel** [comment]
Thanks Sayak!

**Pull request merged**